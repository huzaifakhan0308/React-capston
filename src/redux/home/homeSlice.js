import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';

export const getLatLon = createAsyncThunk(
  'city/LatLon',
  async (cityName) => {
    const response = await axios.get(`http://api.openweathermap.org/geo/1.0/direct?q=${cityName}&limit=5&appid=f617f62839e8a87840642b3875ecf15e`);
    if (response.data[0] === undefined) {
      return { error: 'please correct your city name !' };
    }
    return response.data[0];
  },
);

const homeSlice = createSlice({
  name: 'home',
  initialState: {},
  reducers: {},
  extraReducers(builder) {
    builder.addCase(getLatLon.fulfilled, (state, action) => action.payload);
  },
});

export default homeSlice;
