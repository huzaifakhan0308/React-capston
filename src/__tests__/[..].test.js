import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Navbar from '../components/navbar';
import Home from '../components/home';
import Bydefault from '../components/byDefault';
import Detail from '../components/detail';
import homeSlice from '../redux/home/homeSlice';
import deatilSlice from '../redux/detail/detailSlice';

const store = configureStore({
  reducer: {
    home: homeSlice.reducer,
    detail: deatilSlice.reducer,
  },
});

describe('All components tests', () => {
  it('Navbar component', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Router>
          <Navbar />
        </Router>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Home component', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Router>
          <Home />
        </Router>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Bydefault component', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Router>
          <Bydefault />
        </Router>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Detail component', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Router>
          <Detail />
        </Router>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
