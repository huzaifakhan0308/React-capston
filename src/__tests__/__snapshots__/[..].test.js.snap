// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All components tests Bydefault component 1`] = `
<div
  className="byDefault-container"
>
  <div
    className="byDefault-cities-container"
  >
    <div
      className="byDefault-cities"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      style={
        Object {
          "background": "rgba(0, 0, 0, 0.11)",
        }
      }
      tabIndex={0}
    >
      <img
        alt="weather-icon"
        className="default-weatherIcon"
        src="weatherIcon.png"
      />
      <div
        className=""
      >
        <p
          style={
            Object {
              "fontSize": "19px",
              "fontWeight": "bolder",
            }
          }
        >
          Country: FR,
        </p>
        <p>
          City: Paris
        </p>
        <p>
          Timezone: 3600
        </p>
      </div>
      <img
        alt="next arrow icon"
        className="default-nextArrow"
        src="next.png"
      />
    </div>
    <div
      className="byDefault-cities"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <img
        alt="weather-icon"
        className="default-weatherIcon"
        src="weatherIcon.png"
      />
      <div
        className=""
      >
        <p
          style={
            Object {
              "fontSize": "19px",
              "fontWeight": "bolder",
            }
          }
        >
          Country: US
        </p>
        <p>
          City: New York
        </p>
        <p>
          Timezone: -14400
        </p>
      </div>
      <img
        alt="next arrow icon"
        className="default-nextArrow"
        src="next.png"
      />
    </div>
  </div>
  <div
    className="byDefault-cities-container"
  >
    <div
      className="byDefault-cities"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <img
        alt="weather-icon"
        className="default-weatherIcon"
        src="weatherIcon.png"
      />
      <div
        className=""
      >
        <p
          style={
            Object {
              "fontSize": "19px",
              "fontWeight": "bolder",
            }
          }
        >
          Country: RU
        </p>
        <p>
          City: Moscow
        </p>
        <p>
          Timezone: 10800
        </p>
      </div>
      <img
        alt="next arrow icon"
        className="default-nextArrow"
        src="next.png"
      />
    </div>
    <div
      className="byDefault-cities"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      style={
        Object {
          "background": "rgba(0, 0, 0, 0.11)",
        }
      }
      tabIndex={0}
    >
      <img
        alt="weather-icon"
        className="default-weatherIcon"
        src="weatherIcon.png"
      />
      <div
        className=""
      >
        <p
          style={
            Object {
              "fontSize": "19px",
              "fontWeight": "bolder",
            }
          }
        >
          Country: CN
        </p>
        <p>
          City: Beijing
        </p>
        <p>
          Timezone: 28800
        </p>
      </div>
      <img
        alt="next arrow icon"
        className="default-nextArrow"
        src="next.png"
      />
    </div>
  </div>
  <div
    className="byDefault-cities-container"
  >
    <div
      className="byDefault-cities"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      style={
        Object {
          "background": "rgba(0, 0, 0, 0.11)",
        }
      }
      tabIndex={0}
    >
      <img
        alt="weather-icon"
        className="default-weatherIcon"
        src="weatherIcon.png"
      />
      <div
        className=""
      >
        <p
          style={
            Object {
              "fontSize": "19px",
              "fontWeight": "bolder",
            }
          }
        >
          Country: AE
        </p>
        <p>
          City: Dubai
        </p>
        <p>
          Timezone: 14400
        </p>
      </div>
      <img
        alt="next arrow icon"
        className="default-nextArrow"
        src="next.png"
      />
    </div>
    <div
      className="byDefault-cities"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <img
        alt="weather-icon"
        className="default-weatherIcon"
        src="weatherIcon.png"
      />
      <div
        className=""
      >
        <p
          style={
            Object {
              "fontSize": "19px",
              "fontWeight": "bolder",
            }
          }
        >
          Country: TR
        </p>
        <p>
          City: Istanbul
        </p>
        <p>
          Timezone: 10800
        </p>
      </div>
      <img
        alt="next arrow icon"
        className="default-nextArrow"
        src="next.png"
      />
    </div>
  </div>
</div>
`;

exports[`All components tests Detail component 1`] = `
<div
  className="detail-container"
>
  <div
    className="weather-info-container"
    style={
      Object {
        "color": "white",
      }
    }
  >
    loading....
  </div>
</div>
`;

exports[`All components tests Home component 1`] = `
<div
  className="home-container"
>
  <div
    className="form-container"
  >
    <form
      action=""
    >
      <input
        className="search-city-input"
        onChange={[Function]}
        placeholder="search your city"
        type="text"
        value=""
      />
      <button
        onClick={[Function]}
        type="button"
      >
        <img
          alt="search icon"
          src="search-icon.png"
        />
      </button>
    </form>
  </div>
  <a
    href="/detail"
    onClick={[Function]}
  >
    <div
      className="weather-info-container"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <img
        alt=""
        src="weatherIcon.png"
      />
      <div
        className="weather-info-div"
      >
        <p
          style={
            Object {
              "fontSize": "22px",
              "fontWeight": "bolder",
            }
          }
        >
          Country: PK
        </p>
        <p>
          City: Nowshera
        </p>
        <p>
          Latitude: 34.0156
        </p>
        <p>
          Longitude: 71.9813
        </p>
        <p>
          Timezone: 18000
        </p>
      </div>
      <img
        alt="next arrow"
        className="nextArrow"
        src="next.png"
      />
    </div>
    <div
      className="byDefault-container"
    >
      <div
        className="byDefault-cities-container"
      >
        <div
          className="byDefault-cities"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          style={
            Object {
              "background": "rgba(0, 0, 0, 0.11)",
            }
          }
          tabIndex={0}
        >
          <img
            alt="weather-icon"
            className="default-weatherIcon"
            src="weatherIcon.png"
          />
          <div
            className=""
          >
            <p
              style={
                Object {
                  "fontSize": "19px",
                  "fontWeight": "bolder",
                }
              }
            >
              Country: FR,
            </p>
            <p>
              City: Paris
            </p>
            <p>
              Timezone: 3600
            </p>
          </div>
          <img
            alt="next arrow icon"
            className="default-nextArrow"
            src="next.png"
          />
        </div>
        <div
          className="byDefault-cities"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          <img
            alt="weather-icon"
            className="default-weatherIcon"
            src="weatherIcon.png"
          />
          <div
            className=""
          >
            <p
              style={
                Object {
                  "fontSize": "19px",
                  "fontWeight": "bolder",
                }
              }
            >
              Country: US
            </p>
            <p>
              City: New York
            </p>
            <p>
              Timezone: -14400
            </p>
          </div>
          <img
            alt="next arrow icon"
            className="default-nextArrow"
            src="next.png"
          />
        </div>
      </div>
      <div
        className="byDefault-cities-container"
      >
        <div
          className="byDefault-cities"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          <img
            alt="weather-icon"
            className="default-weatherIcon"
            src="weatherIcon.png"
          />
          <div
            className=""
          >
            <p
              style={
                Object {
                  "fontSize": "19px",
                  "fontWeight": "bolder",
                }
              }
            >
              Country: RU
            </p>
            <p>
              City: Moscow
            </p>
            <p>
              Timezone: 10800
            </p>
          </div>
          <img
            alt="next arrow icon"
            className="default-nextArrow"
            src="next.png"
          />
        </div>
        <div
          className="byDefault-cities"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          style={
            Object {
              "background": "rgba(0, 0, 0, 0.11)",
            }
          }
          tabIndex={0}
        >
          <img
            alt="weather-icon"
            className="default-weatherIcon"
            src="weatherIcon.png"
          />
          <div
            className=""
          >
            <p
              style={
                Object {
                  "fontSize": "19px",
                  "fontWeight": "bolder",
                }
              }
            >
              Country: CN
            </p>
            <p>
              City: Beijing
            </p>
            <p>
              Timezone: 28800
            </p>
          </div>
          <img
            alt="next arrow icon"
            className="default-nextArrow"
            src="next.png"
          />
        </div>
      </div>
      <div
        className="byDefault-cities-container"
      >
        <div
          className="byDefault-cities"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          style={
            Object {
              "background": "rgba(0, 0, 0, 0.11)",
            }
          }
          tabIndex={0}
        >
          <img
            alt="weather-icon"
            className="default-weatherIcon"
            src="weatherIcon.png"
          />
          <div
            className=""
          >
            <p
              style={
                Object {
                  "fontSize": "19px",
                  "fontWeight": "bolder",
                }
              }
            >
              Country: AE
            </p>
            <p>
              City: Dubai
            </p>
            <p>
              Timezone: 14400
            </p>
          </div>
          <img
            alt="next arrow icon"
            className="default-nextArrow"
            src="next.png"
          />
        </div>
        <div
          className="byDefault-cities"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          <img
            alt="weather-icon"
            className="default-weatherIcon"
            src="weatherIcon.png"
          />
          <div
            className=""
          >
            <p
              style={
                Object {
                  "fontSize": "19px",
                  "fontWeight": "bolder",
                }
              }
            >
              Country: TR
            </p>
            <p>
              City: Istanbul
            </p>
            <p>
              Timezone: 10800
            </p>
          </div>
          <img
            alt="next arrow icon"
            className="default-nextArrow"
            src="next.png"
          />
        </div>
      </div>
    </div>
  </a>
</div>
`;

exports[`All components tests Navbar component 1`] = `
<nav>
  <a
    href="/"
    onClick={[Function]}
  />
  <span
    className="page-names"
  >
    cities
  </span>
  <div
    className="mic-setting-icon-div"
  >
    <img
      alt="mic icon"
      className="micIcon"
      src="mic.png"
    />
    <img
      alt="setting icon"
      className="settingIcon"
      src="gear.png"
    />
  </div>
</nav>
`;
